### PR TITLE
Regression 3.7: last template style preview not working #1

### PR DESCRIPTION
**Pull request by joeforjoomla**
Reference https://github.com/joomla/joomla-cms/issues/13637

### Steps to reproduce the issue
Install a new template on Joomla 3.7

Try to open the preview using the URL such as:
http://joomla37/?template=mytemplate

### Expected result
The frontend uses the latest installed template named 'mytemplate'

### Actual result
The default template is still used

### System information (as much as possible)
The code change in site.php included in the function 'getTemplate' caused the regression.

An additional line of code 510 must be added in order to unset the reference to the array before reassigning it:

// Unset the $template reference to the last $templates[n] item cycled in the foreach above to avoid to edit the $templates array in the following assignment 
unset($template);

### Additional comments

### CODE REVIEW

**Comment by infograf768** [comment]
you are modifying parts of this pr https://github.com/joomla/joomla-cms/pull/12688 please make sure you do not recreate the issue it solved.

**Comment by joeforjoomla** [comment]
Probably it would be better fix the new code #12688 instead of reverting to the previous one that i posted

**Comment by mbabker** [comment]
It looks like you are reverting changes from several pull requests and moving back to the 3.6.5 state.  Please don't do that, only make the changes needed to fix the issue.  As [I commented](https://github.com/joomla/joomla-cms/issues/13637#issuecomment-273802942) on your issue last week that function has had four separate pull requests applied to it when comparing 3.6.5 to current staging.

**Comment by joeforjoomla** [comment]
Add line 510 

`// Unset the $template reference to the last $templates[n] item cycled in the foreach above to avoid to edit the $templates array in the following assignment 
		unset($template);`

File: libraries/cms/application/site.php [hunk 1]
```diff
@@ -507,6 +507,8 @@ public function getTemplate($params = false)
 			$cache->store($templates, $cacheId);
 		}
 
+		// Unset the $template reference to the last $templates[n] item cycled in the foreach above to avoid to edit the $templates array in the following assignment
```
Comment (mbabker) [inline libraries/cms/application/site.php]: This should be done in the else part of the above if/else.  `$template` isn't defined if the data is pulled out of the cache.

**Comment by mbabker** [comment]
OK, I moved the check myself.  This PR looks fine to me, tests please.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 796a05781a6c11b125351993d403917165edb65d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13775">issues.joomla.org/tracker/joomla-cms/13775</a>.</sub>

**Comment by marrouchi** [comment]
I have tested this item :white_check_mark: successfully on 796a05781a6c11b125351993d403917165edb65d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13775">issues.joomla.org/tracker/joomla-cms/13775</a>.</sub>

**Comment by infograf768** [comment]
RTC. Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13775">issues.joomla.org/tracker/joomla-cms/13775</a>.</sub>

**Pull request merged**